### PR TITLE
[JAVA-9317] Update tests to use JsonUnit for improved assertions

### PR DESCRIPTION
**Pull request by hkhan**

### CODE REVIEW

File: json/src/test/java/com/baeldung/jsonjava/CDLIntegrationTest.java [hunk 1]
```diff
@@ -1,36 +1,44 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+
 public class CDLIntegrationTest {
+
     @Test
     public void givenCommaDelimitedText_thenConvertToJSONArray() {
         JSONArray ja = CDL.rowToJSONArray(new JSONTokener("England, USA, Canada"));
-        assertEquals("[\"England\",\"USA\",\"Canada\"]", ja.toString());
+
+        assertThatJson(ja.toString())
```
Comment (kwoyke) [inline json/src/test/java/com/baeldung/jsonjava/CDLIntegrationTest.java:17]: Should we really use toString here?

File: json/pom.xml [hunk 2]
```diff
@@ -63,6 +63,12 @@
             <version>${commons-collections4.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>net.javacrumbs.json-unit</groupId>
+            <artifactId>json-unit-assertj</artifactId>
+            <version>2.28.0</version>
```
Comment (kwoyke) [inline json/pom.xml:69]: Pls move the version to property.

File: json/src/test/java/com/baeldung/jsonjava/CookieIntegrationTest.java [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
 
 public class CookieIntegrationTest {
+
     @Test
     public void givenCookieString_thenConvertToJSONObject() {
         String cookie = "username=John Doe; expires=Thu, 18 Dec 2013 12:00:00 UTC; path=/";
         JSONObject cookieJO = Cookie.toJSONObject(cookie);
 
-        assertEquals("{\"path\":\"/\",\"expires\":\"Thu, 18 Dec 2013 12:00:00 UTC\",\"name\":\"username\",\"value\":\"John Doe\"}", cookieJO.toString());
+        assertThatJson(cookieJO.toString())
```
Comment (kwoyke) [inline json/src/test/java/com/baeldung/jsonjava/CookieIntegrationTest.java:17]: toString ?

File: json/src/test/java/com/baeldung/jsonjava/CDLIntegrationTest.java (hunk 1 above)
Comment (hkhan) [reply json/src/test/java/com/baeldung/jsonjava/CDLIntegrationTest.java]: shouldn't need toString here - fixed

File: json/src/test/java/com/baeldung/jsonjava/CookieIntegrationTest.java (hunk 3 above)
Comment (hkhan) [reply json/src/test/java/com/baeldung/jsonjava/CookieIntegrationTest.java]: not needed

**Pull request merged**